src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx

```diff
@@ -0,0 +1,107 @@
+import { SelectableValue, toOption } from '@grafana/data';
+import React, { useState } from 'react';
+
+import { InlineField, MultiSelect, Select } from '@grafana/ui';
+import _ from 'lodash';
+import { useAsync } from 'react-use';
+import { FloatingError, parseErrorMessage } from '../../../../../core/errors';
+import { useWorkspaceOptions } from '../../../../../core/utils';
+import { isValidId } from '../../../../data-frame/utils';
+import { SystemMetadata } from '../../../../system/types';
+import { AssetQuery, ListAssetsQuery } from '../../../types';
+import { ListAssetsDataSource } from './ListAssetsDataSource';
+
+type Props = {
+  query: ListAssetsQuery;
+  handleQueryChange: (value: AssetQuery, runQuery: boolean) => void;
+  datasource: ListAssetsDataSource;
+};
+
+export function ListAssetsEditor({ query, handleQueryChange, datasource }: Props) {
+  query = datasource.prepareQuery(query) as ListAssetsQuery;
+
+  const workspaces = useWorkspaceOptions(datasource);
+  const [errorMsg, setErrorMsg] = useState<string | undefined>('');
+  const handleError = (error: Error) => setErrorMsg(parseErrorMessage(error));
+
+  const minionIds = useAsync(() => {
+    let filterString = '';
+    if (query.workspace) {
+      filterString += `workspace = "${query.workspace}"`;
+    }
+    return datasource.querySystems(filterString).catch(handleError);
+  }, [query.workspace]);
+
+  const onWorkspaceChange = (item?: SelectableValue<string>): void => {
+    if (item?.value && item.value !== query.workspace) {
+      // if workspace changed, reset Systems and Assets fields
+      handleQueryChange(
+        { ...query, workspace: item.value, minionIds: [] },
+        // do not run query if workspace not changed
+        true
+      );
+    } else {
+      handleQueryChange({ ...query, workspace: '' }, true);
+    }
+  };
+
+  const handleMinionIdChange = (items: Array<SelectableValue<string>>): void => {
+    if (items && !_.isEqual(query.minionIds, items)) {
+      handleQueryChange(
+        { ...query, minionIds: items.map(i => i.value!) },
+        // do not run query if minionIds not changed
+        true
+      );
+    } else {
+      handleQueryChange({ ...query, minionIds: [] }, true);
+    }
+  };
+
+  const getVariableOptions = (): Array<SelectableValue<string>> => {
```

Comment:
we can leave this as is for now but I'll reach out to Josh to ask about this. I'm not sure if we should also include the variables as values in the system dropdown.

This is most likely going to be impacted by the query builder anyways.